### ISSUE TITLE
[2.1] & [3.0]: Wrong alert message upon user activation

### ISSUE THREAD

**Issue by sbulen** [SimpleMachines/SMF#8305]
### Basic Information

This is a nit, but a pretty annoying one if you require admin approval & user activation.  I get several of these erroneous alerts a day.

The alert when the new user activates their account uses the text: "xxxx just signed up (account requires activation)".

But they don't need activation...  This was sent when they activated...

I don't know if this is a logic error (reusing message inappropriately) or a simple text error (text needs to change).  

I've confirmed the same issue exists in 2.1 and 3.0.

### Steps to reproduce

1.  Config so new users require admin approval & email activation.
2.  Test user signs up.
3.  Admin approves (requiring email activation).
4. Test user receives activation email
5. Test user clicks on link, gets brought to SMF logon screen, is informed account has been activated.  (so far, so good, but...)
6. Admin receives an alert saying that the test user just signed up & the account needs activation...  

### Expected result

The alert should say that the user activated their account.

(Or, possibly, not send an alert at all???)

### Actual result

AFTER user activation, an alert saying they need to be activated is sent:
![image](https://github.com/user-attachments/assets/e47bed26-284d-430d-8677-2acdb87e2c51)

### Version/Git revision

3.0 alpha 2 & 2.1.4

### Database Engine

All

### Database Version

8.4

### PHP Version

8.3.8

**Comment by jdarwood007** [comment]
I'm having trouble reproducing.  I think you're seeing a delay here.

The `Register_Notify_Background` task is what sends out notifications/emails for this.

Searching the code, this is the only section of code that sets that up
https://github.com/SimpleMachines/SMF/blob/951cc448fb0f65f131cd00b923b8d1bb12b110b7/Sources/Subs-Post.php#L2939-L2949

Searching the code again for `adminNotify` yields a few places
https://github.com/SimpleMachines/SMF/blob/951cc448fb0f65f131cd00b923b8d1bb12b110b7/Sources/Register.php#L734
https://github.com/SimpleMachines/SMF/blob/951cc448fb0f65f131cd00b923b8d1bb12b110b7/Sources/Subs-Members.php#L801
https://github.com/SimpleMachines/SMF/blob/951cc448fb0f65f131cd00b923b8d1bb12b110b7/Sources/Subs-Members.php#L844

The last one is the only one to send the `approval` template.  This code is only triggered when they first register.  It doesn't trigger on the admin requiring them to activate and sending them an email.  Nor does it trigger when they activate.

**Comment by sbulen** [comment]
Don't know what to tell you.  I get these every single day in production...   And I have for years...  

And I can get them in test following the steps above in both vanilla 2.1 & 3.0.

I did make a minor tweak to the steps to reproduce above - that requiring activation is selected by the admin upon approval (not configured beforehand).

Haven't looked at the code to debug.

**Comment by jdarwood007** [comment]
The problem is, I am not certain how you are getting it then.  The code only shows 1 place where we would send the register_approval notification template, and that's the path I mentioned.

**Comment by sbulen** [comment]
Have you followed the steps above?

**Comment by jdarwood007** [comment]
Yes, I did.  I got the notification after the user registered but before the admin approved.  Admin approved and then the account was activated and no notification.
Do you get the notification on the user registration as well (but before you activate)?

**Comment by sbulen** [comment]
Yes, as shown in the screenshot above.  Two notifications - one upon request, indicating "account requires approval".  The next, after approval (via "approve requiring activation"), after the user clicks on the activation link, indicates "account requires activation".  

Two different notifications.  Two different notification messages.  As in the screenshot above.  The second one is inaccurate, as it indicates the account requires activation.  

EDIT:  It's possible the second one is triggered by the admin approval (not the user hitting the link)???

**Comment by jdarwood007** [comment]
I wasn't getting it when the admin approved.  That should be fairly easy to see if you get a notification when another admin approves them.

What I did after not reproducing it is work backwards.  The user_alert has for the content_type, 'register_approval'.  I checked backwards in the code and couldn't find the exact match, but splitting the strings is where i verified what I saw.

That is what is confusing me, is there should be no way to get what your seeing.

I didn't dig the entire code, but the activate section is the next place to fully search.  Which is what activates the account from the link the user is sent.  That is part of the first match.  But that should be register_activation.

Do we perhaps have a language string issue?  Can you confirm what the database has for the user_alert?

**Comment by sbulen** [comment]
Yep, Register_Notify_Background() is called.  And it's the adminNotify('activation'...) that is called when the user clicks on their email activation link.

The issue is in the Register-Notify task here, which is called to tell folks the user was just activated:
https://github.com/SimpleMachines/SMF/blob/c804486ccea47f6e137cb38e41d6073557e31d06/Sources/tasks/Register-Notify.php#L67

...since the notify_type is 'activation', it will build the key for the alert message as: 'register_activation'

Which ends up resolving to:
https://github.com/SimpleMachines/SMF/blob/c804486ccea47f6e137cb38e41d6073557e31d06/Themes/default/languages/Alerts.english.php#L35

Which is the erroneous message...  The user was just activated...

I've confirmed the alert is incorrect both when the registration is 'Email Activation' (no admin intervention required) and when the registration is 'Admin Approval' & the admin chooses 'Approve and require activation' upon approval (my site's process).

Under both scenarios, the 'requires activation' alert is sent **_after_** the user is activated.

I'm hesitant to just change the message, because I'm unsure if that particular message is appropriate for some other scenario...

Just a nit, but I get 5-10 of these throughout the day...  It's like... water torture...

**Comment by jdarwood007** [comment]
https://github.com/SimpleMachines/SMF/blob/951cc448fb0f65f131cd00b923b8d1bb12b110b7/Sources/Register.php#L724-L735

The process that gets triggered is updating the member data to activate them, then checking the old status.  It was 0 so we sent the notification to the admin.

Perhaps we just need to change the language string to say `account was activated`?

**Comment by sbulen** [comment]
Yep.

**Comment by sbulen** [comment]
Only concern here is the message as-is may be valid in a similar flow.  I'll do more testing.

**Comment by jdarwood007** [comment]
@sbulen Any updates here yet?
It appears that the fix involves updating the language string to indicate the past-tense nature of the message.

**Comment by sbulen** [comment]
No, I haven't come back to this one yet.

The one chink in the armor that I am concerned about is whether there is another place where that string is used appropriately.  I was hoping to poke around & confirm or deny that.  If so, we'd need a new string.  If not, we can change the existing one.
